RPI2/Ubuntu: add uhubctl to restart USB Hub

rmackay9 opened the pull request:
This PR corrects a few issues with the RPI2/Ubuntu scripts:

- install scripts executable permissions fixed
- adds uhubctl so that RPI4's USB ports are powered off/on to allow the T265 to start without user intervention
- simplifies the human readable install instructions to point users at the automated scripts
- removes swap file creation which allows the scripts to be run on an RPI4 with an 8GB SD card
- separates the install_librealsense and instasll_vision_to_mavros so they can be run separately
- adds clean-for-imaging script which is creates an image of the SD card when run on a PC (I think)

rmackay9 committed:
Common/Ubuntu: add uhubctl install and start scripts

rmackay9 committed:
RPI2/Ubuntu: add clean-for-imaging

rmackay9 committed:
RPI2/Ubuntu: all scripts made executable

rmackay9 committed:
Common/Ubuntu: fix buildLibrealsense.sh script

rmackay9 committed:
RPI2/Ubuntu: 4_setup_apsync_components loses swap file creation

also remove commented out lines

rmackay9 committed:
RPI2/Ubuntu: simplify instructions to always use automated scripts

rmackay9 committed:
RPI2/Ubuntu: add 6_setup_uhubctl.sh

rmackay9 committed:
Common/Ubuntu: librealsense outputs start and end msg

also update usage in advance of renaming script

rmackay9 committed:
Common/Ubuntu: rename buildLibrealsense to install_librealsense

rmackay9 committed:
Common/Ubuntu: separate instal_librealsense and vision_to_mavros

rmackay9 committed:
RPI2/Ubuntu: install_realsense.sh calls install_librealsense and install_vision_to_mavros individually

rmackay9 committed:
Common/Ubuntu: install_vision_to_mavros.sh removes prompt to install

rmackay9 committed:
Common/Ubuntu: add stop and viewlog for t265 service

rmackay9 committed:
Common/Ubuntu: fix install_librealsense

rmackay9 committed:
Common/Ubuntu: install pip3 and future

rmackay9 committed:
RPI2/Ubuntu: setup_master_mavlink-router removes /etc/mavlink-router directory

also remove install of future because these are now in the Common/Ubuntu install-packages script

rmackay9 committed:
RPI2/Ubuntu: 4_setup_apsync_components does not install future

these are now in the Common/Ubuntu install-packages script

khancyr commented on Common/Ubuntu/librealsense/stop_t265.sh:
```diff
@@ -0,0 +1,10 @@
+   exit 1
+fi
+
+sudo systemctl stop t265
```
nitpick, if you are root, you don't need sudo

rmackay9 replied:
Thanks, fixed!

khancyr commented on RPI2/Ubuntu/clean-for-imaging:
```diff
@@ -0,0 +1,22 @@
+
+rm -f /home/$NORMAL_USER/.ssh/authorized_keys
+
+dd if=/dev/zero of=$HOME/zeroes || true
```
dd if=/dev/zero of=$HOME/zeroes status=progress

rmackay9 replied:
This part is from @peterbarker so I'll let him comment..

peterbarker replied:
@rmackay9 this just adds a comforter so you know things are moving along.

Wouldn't hurt.

rmackay9 committed:
Common/Ubuntu: fixup librealsense scripts

add autostart of t265 service and add to /etc/rc.local
remove unnecessary sudo
remove some directories to make it easier for script to be reun

rmackay9 commented:
I think these scripts work now from start to finish.  I ran though a full install today but bumped into a few issues.  I've fixed those and I will try the full procedure again tomorrow.

rmackay9 commented:
I've tested these now from start to finish and it comes up with a working image in which the Copter appears in Africa without the user having to make any changes on the RPI4.  I'll go ahead and merge this soon and then hopefully get an image onto firmware.ardupilot.org.
